### ISSUE TITLE
ArcGIS Enterprise HA in premises: DataStore "EnableFailoverOnPrimaryStop" property not working

### ISSUE THREAD

**Issue by Biboba** [Esri/arcgis-powershell-dsc#313]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.1.1

### Affected Resource(s)

* ArcGIS_DataStore

### Configuration Files

[ArcGIS_Enterprise_HA.txt](https://github.com/Esri/arcgis-powershell-dsc/files/6184523/ArcGIS_Enterprise_HA.txt)

### Expected Behavior

As "EnableFailoverOnPrimaryStop" property is to true in the datastoreconfig, I would expect to have the property 'failover_on_primary_stop' in the file C:\Program Files\ArcGIS\DataStore\framework\etc\datastore.properties set to true
```
# When failover_on_primary_stop is set true, failover will take place when the primary
# machine is stopped. Default is false.
failover_on_primary_stop=true
```

### Actual Behavior

The property 'failover_on_primary_stop' in the file C:\Program Files\ArcGIS\DataStore\framework\etc\datastore.properties was  not set to true
```
# When failover_on_primary_stop is set true, failover will take place when the primary
# machine is stopped. Default is false.
failover_on_primary_stop=false
```

Also nothing is logged in the log regarding the testing of this property

### Steps to Reproduce

Run the attached config. Once over, check the file C:\Program Files\ArcGIS\DataStore\framework\etc\datastore.properties in datastore install folder. Property 'failover_on_primary_stop' is set to false

### Important Factoids

Looking at the source code, I understood that this property is only applied in azure env:
https://github.com/Esri/arcgis-powershell-dsc/blob/bd004e9b12fe31bf47918a9c1cf690e95640f414/Modules/ArcGIS/DSCResources/ArcGIS_DataStore/ArcGIS_DataStore.psm1#L157

In the [Wiki](https://github.com/Esri/arcgis-powershell-dsc/wiki/V3.-Variables-reference-page-for-JSON-configuration-files), it is not mentionned that this property is only working Azur:

```
EnableFailoverOnPrimaryStop - (Optional) Boolean (true|false). Set to true to enable failover to secondary when primary datastore is stopped. The default is false.
```

Is it not supported for on premises deployment ?

I tried to disable this check but it did not work. Looking at the code, it seems like once again, you set the property only on Azure:
https://github.com/Esri/arcgis-powershell-dsc/blob/bd004e9b12fe31bf47918a9c1cf690e95640f414/Modules/ArcGIS/DSCResources/ArcGIS_DataStore/ArcGIS_DataStore.psm1#L177

### References

Details explanations about this property in https://github.com/Esri/arcgis-powershell-dsc/issues/261

Thanks for your help !

**Comment by Biboba** [comment]
Seems like this azur condition is also mentionned here:
https://github.com/Esri/arcgis-powershell-dsc/issues/226

**Comment by cameronkroeker** [comment]
Hi @Biboba,

This is a bug as it should work in both on-prem and cloud deployments. We will address this in the next release, in the mean time feel free to use the following workaround:

**Replace** 
`$ConfigurationParamsHashtable.ConfigData.DataStoreItems.DataStore.EnableFailoverOnPrimaryStop`

**with**

`$ConfigurationParamsHashtable.ConfigData.DataStore.EnableFailoverOnPrimaryStop`

**In the following 3 locations:**

1. https://github.com/Esri/arcgis-powershell-dsc/blob/bd004e9b12fe31bf47918a9c1cf690e95640f414/Modules/ArcGIS/ArcGIS.psm1#L1351
2. https://github.com/Esri/arcgis-powershell-dsc/blob/bd004e9b12fe31bf47918a9c1cf690e95640f414/Modules/ArcGIS/ArcGIS.psm1#L1369
3. https://github.com/Esri/arcgis-powershell-dsc/blob/bd004e9b12fe31bf47918a9c1cf690e95640f414/Modules/ArcGIS/ArcGIS.psm1#L1333

Thanks,
Cameron K.

**Comment by Biboba** [comment]
Thanks for the information @cameronkroeker  !

I just tested your workaround but it does not work on my side.
What worked though is to ignore this test with (if $true)
https://github.com/Esri/arcgis-powershell-dsc/blob/bd004e9b12fe31bf47918a9c1cf690e95640f414/Modules/ArcGIS/DSCResources/ArcGIS_DataStore/ArcGIS_DataStore.psm1#L157

and to replace the following:
https://github.com/Esri/arcgis-powershell-dsc/blob/bd004e9b12fe31bf47918a9c1cf690e95640f414/Modules/ArcGIS/DSCResources/ArcGIS_DataStore/ArcGIS_DataStore.psm1#L177

with 

```
if(Set-PropertyFromPropertiesFile -PropertiesFilePath $PropertiesFilePath -PropertyName $PropertyName -PropertyValue 'true' -Verbose){ 
```

Dirty, but it works. Just ignoring this IsEnvAzure parameter

**Comment by cameronkroeker** [comment]
Hi @Biboba,

This issue has been addressed in the latest release, v3.2.0:

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v3.2.0

Thanks,
Cameron K.
